baseplate/sidecars/live_data_watcher.py

```diff
@@ -32,6 +37,64 @@ def __init__(self, dest: str, owner: int, group: int, mode: int):
         self.group = group
         self.mode = mode
 
+    @staticmethod
+    def get_encrypted_json_from_s3(
+        bucket_name: str, file_key: str, region_name: str, sse_key: str
+    ) -> Optional[bytes]:
+        s3_client = boto3.client(
+            "s3",
+            region_name=region_name,
+        )
+        data: Optional[bytes] = None
+        # The S3 data must be Server Side Encrypted and we should have the decryption key.
+        kwargs = {
+            "Bucket": bucket_name,
+            "Key": file_key,
+            "SSECustomerKey": sse_key,
```

Comment:
looking at boto's doc it's not 100% clear to me whether this "key" is supposed to be the whole secret or a "key" to look up the secret from some storage.

also looking at https://docs.aws.amazon.com/AmazonS3/latest/userguide/ServerSideEncryptionCustomerKeys.html, it seems that using SSE means the decryption actually happens on server side, so the data transferred between the wire is the already decrypted data (but of course the transfer is still encrypted by tls). I'm not sure if that's the preferred the way to do that, maybe we would prefer to manage the encryption/decryption entirely on the client side?

Reply:
To your first question - I verified this in our service code and it's the whole secret. In experiment-config service we upload to our bucket with this encryption so the file stored there is completely SSE encrypted.

In the console:

![Screen Shot 2022-04-08 at 10 31 06 AM](https://user-images.githubusercontent.com/5678562/162492435-f96f4d3e-a88b-4ec2-8622-a43cb78f114d.png)

I also verified that trying to decrypt this using a faulty key will not work. I received AccessDenied Error.

Reply:
cc @sp3nx0r on this thread

Reply:
Also errors when trying to download it (since it's encrypted):

![Screen Shot 2022-04-08 at 10 52 34 AM](https://user-images.githubusercontent.com/5678562/162495443-9ccdd7a5-b125-4225-813e-c9170cebd443.png)

Reply:
alternative would be [client side encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingClientSideEncryption.html). Since it's still TLS wrapped, the pull and decrypt security steps should be sufficient for what we're trying to protect here.